src/quo2/components/messages/author/style.cljs

```diff
@@ -2,30 +2,44 @@
   (:require [quo2.foundations.colors :as colors]))
 
 (def container
-  {:flex-wrap      :wrap
+  {:flex-wrap      :nowrap
    :flex-direction :row
    :align-items    :center})
 
 (defn middle-dot-nickname
-  []
-  {:color             (colors/theme-colors colors/neutral-40 colors/neutral-50)
+  [theme]
+  {:color             (colors/theme-colors colors/neutral-40 colors/neutral-50 theme)
    :margin-horizontal 4})
 
 (defn chat-key-text
-  []
-  {:color       (colors/theme-colors colors/neutral-40 colors/neutral-50)
+  [theme]
+  {:color       (colors/theme-colors colors/neutral-40 colors/neutral-50 theme)
    :margin-left 8})
 
 (defn middle-dot-chat-key
-  []
-  {:color       (colors/theme-colors colors/neutral-40 colors/neutral-50)
+  [theme]
+  {:color       (colors/theme-colors colors/neutral-40 colors/neutral-50 theme)
    :margin-left 4})
 
+(defn primary-name
+  [muted? theme]
+  {:color       (if muted?
+                  colors/neutral-50
+                  (colors/theme-colors colors/neutral-100 colors/white theme))
+   :flex-shrink 1})
+
+(defn secondary-name
+  [theme]
+  {:padding-top 1
+   :flex-shrink 999999
```

Comment:
The requirement is to truncate the nickname(secondary-name) and stop truncation when it reaches a width of 40px. Subsequently, we begin truncating the name. While this approach may not be flawless, setting a high flex-shrink value causes the nickname to shrink at a faster rate. Based on my testing, this results in the name starting to truncate when the nickname reaches the min-width(40px).

If you have any alternative suggestions to address this issue, please feel free to share.